presentation/src/main/java/com/ragdroid/mvi/presenters/MainFragmentPresenterImpl.kt

```diff
@@ -0,0 +1,92 @@
+package com.ragdroid.mvi.presenters
+
+import com.ragdroid.data.MainRepository
+import com.ragdroid.data.base.SchedulerProvider
+import com.ragdroid.mvi.main.MainFragmentPresenter
+import com.ragdroid.mvi.main.MainFragmentView
+import com.ragdroid.mvi.main.MainResult
+import com.ragdroid.mvi.main.MainViewState
+import io.reactivex.Observable
+import io.reactivex.disposables.Disposable
+import io.reactivex.subjects.PublishSubject
+import timber.log.Timber
+import javax.inject.Inject
+
+/**
+ * Created by garimajain on 19/11/17.
+ */
+class MainFragmentPresenterImpl
+@Inject constructor(private val repository: MainRepository,
+                    private val schedulerProvider: SchedulerProvider) : MainFragmentPresenter {
+
+    val subject: PublishSubject<MainResult> = PublishSubject.create()
+    var disposable: Disposable? = null
+
+    var view: MainFragmentView? = null
```

Comment:
`view` can be non-null

Reply:
@chetan-fueled I think this `view` should be `nullable` as in `detachView()` we will be setting it to `null` and we don't want to be accessing our view when we come back from another thread and access our `view`.

Here `nullability` can be our friend. I think.